manifests/certonly.pp

```diff
@@ -33,24 +33,22 @@
 #   succeeds.
 #
 define letsencrypt::certonly (
-  Array $domains                                            = [$title],
-  Boolean $custom_plugin                                    = false,
-  Enum['apache', 'standalone', 'webroot', 'nginx', 'dns-route53', 'dns-google'] $plugin  = 'standalone',
-  Optional[Array] $webroot_paths                            = undef,
-  String $letsencrypt_command                               = $letsencrypt::command,
-  Optional[Array] $additional_args                          = undef,
-  Array $environment                                        = [],
-  Boolean $manage_cron                                      = false,
-  Boolean $suppress_cron_output                             = false,
-  $cron_before_command                                      = undef,
-  $cron_success_command                                     = undef,
-  Stdlib::Unixpath $config_dir                              = $letsencrypt::config_dir,
+  Array               $domains              = [$title],
```

Comment:
can you enforce the contents of the arrays as well? Probably ```Array[String[1]]``` for most occasions.

Reply:
Done, but decided to use `Array[Stdlib::Host]` for `$domains` and `Array[Stdlib::Unixpath]` for `$webroot_paths`.